### ISSUE TITLE
iOS - Bookmark - Incorrect Heading Levels

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#97]
When a VoiceOver user goes to the Bookmark screen, the text "Bookmark" is read as a heading level four (h4). I would expect it to be a heading level one (h1) or maybe a heading level two (h2) if there were a heading level one (h1) above it.

Also, why is the page I bookmarked text a heading level three (h3)? I would expect it to be a list item and not a heading.

**Comment by jfc3-dol** [comment]
VoiceOver still reading the text "Bookmark" as heading level four (h4).

**Comment by KyleLivingstonDOL** [comment]
ZTI, can you please address the heading level on the Bookmarks page?

@jfc3-dol is the reading of bookmarks as either H2/h3 versus a list a 508 issue? It is a heading level because -- I believe -- when bookmarking a step and a topic, the STEP name is H2/H3 and the topic is H3/H4. ZTI can you please confirm?

**Comment by jfc3-dol** [comment]
@KyleLivingstonDOL,

The page name "Bookmark should be a heading level one (h1), and it's not at the moment. Also, all the bookmarked items should be a heading level two (h2). If the item has a Step and a Topic, the entire item should be a heading level two (h2).

Please see the attached screenshot of what I'm referring too.

**Screenshot of current heading structure overlaid with what it should be.**

![bookmark - heading structure needs to change](https://user-images.githubusercontent.com/10423487/68400203-a3298580-0145-11ea-988c-104c5231fc2d.png)

**Comment by Abychalil** [comment]
Addressed this issue. Please retest and verify whether it meets the required 508 standards.
@jfc3-dol

**Comment by jfc3-dol** [comment]
@Abychalil ,

In continuous reading mode, VoiceOver is not reading the text "Bookmark" in the main content section of the page as a heading level one (h1) as expected or the bookmarked items. Now, if you read bookmarked items using a one-finger swipe right, VoiceOver is reading the text "bookmark" as a heading level one (h1) as expected. Using a one-finger swipe for any "Step" (black text), VoiceOver reads it as a heading level two (h2), but if an item does not have a "Step", it is not read as a heading, and the blue text is read as a button only.

So this still needs work.

**Comment by jfc3-dol** [comment]
@Abychalil,
The above from February 18th is still true.

VoiceOver is also reading the items as both a button and a link for some reason. I would expect them to be links because they take you to a given page.

**Comment by jfc3-dol** [comment]
@Abychalil ,

In continuous reading mode, VoiceOver is not reading the text "Bookmark" in the main content section of the page as a heading level one (h1) as expected or the bookmarked items. VoiceOver is reading the bookmarked items as a button and a link but not a heading. I think because it's marked as both a button and a link, it might be causing issues for VoiceOver to read them as headings.

Now, if you read bookmarked items using a one-finger swipe right, VoiceOver is reading the text "Bookmark" as a heading level one (h1) as expected. Using a one-finger swipe for any "Step" (black text), VoiceOver reads it as a heading level two (h2), but if an item does not have a "Step", it is not read as a heading, and the blue text is read as a button and a link.

So this still needs work, and I am assuming there is the same issue for the web and Android version too.

I would expect bookmarked items to be read as a heading and a link because it's taking you to a page if you tap on it. That includes both the Step text and the blue text too.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

I had the same experience with VoiceOver today as I did back on July 23, as mentioned above.

**Comment by jfc3-dol** [comment]
The same applies still today as my comment from July 23, 2020.

**Comment by pbhatt17** [comment]
@jfc3-dol 
Please retest, bookmark headings are already corrected already.

**Comment by jfc3-dol** [comment]
@pbhatt17 

VociceOver is now reading both the Step if present and the Topic as one link in continuous reading mode (two-finger down swipe), so users will know which Topic and Step they have bookmarked.

I am closing this issue.
